src/components/EditorMarkdownIt.vue

```diff
@@ -51,7 +55,49 @@ export default {
 	methods: {
 		onUpdate() {
 			this.html = this.md.render(this.value)
+			setTimeout(() => this.prepareOnChangeListener(), 100)
+		},
+
+		prepareOnChangeListener() {
+			if (this.readonly) {
+				return
+			}
+			const items = document.getElementsByClassName('task-list-item')
+			for (let i = 0; i < items.length; ++i) {
+				items[i].removeEventListener('click', this.setListener)
+				items[i].addEventListener('click', this.setListener)
+			}
+		},
+
+		setListener(item) {
+			let idOfCheckbox = 0
+			const markdownLines = this.value.split('\n')
+			markdownLines.forEach((line, i) => {
+				// Regex Source: https://github.com/linsir/markdown-it-task-checkbox/blob/master/index.js#L121
+				// plus the '- '-string.
+				if (/^(- )\[[xX \u00A0]\][ \u00A0]/.test(line.trim())) {
+					markdownLines[i] = this.checkLine(line, i, idOfCheckbox, item.target)
+					idOfCheckbox++
+				}
+			})
+
+			this.$emit('input', markdownLines.join('\n'))
+		},
+
+		checkLine(line, index, id, target) {
+			let returnValue = line
+			if ('cbx_' + id === target.id) {
+				if (target.checked) {
+					returnValue = returnValue.replace('[ ]', '[x]')
+					returnValue = returnValue.replace('[\u00A0]', '[x]')
+				} else {
+					returnValue = returnValue.replace('[x]', '[ ]')
+					returnValue = returnValue.replace('[X]', '[ ]')
+				}
```

Comment:
This does not work correctly if there is a `[ ]` or `[x]` in the item's text, e.g. `- [x] this is an example [x] with checkbox`. In this case, the text is also changed. In order to fix this, it must be assured, that only a single replace is done.

Reply:
As far as i know the .replace() function only replaces the first occurence, when no regex is used. Even if a regex is applied, when the global-flag is missing it will also only chamge the first occurence. So this should be fine, did your test yield different results?

Reply:
Please try my example. It doesn't work correctly. But I did not found out why.

Reply:
It works just fine for me, have you cleared the cache? (I had problems when the developer console wasnt open, it was using some broken hybrid version from cache)

Reply:
It only has the behaviour you describe if we have both an uppercase and a lowercase X in the line. I will have to fix this particular issue though.

Reply:
I've found out that it happens on sub-items only, see the following example:
 ```
- [ ] check [ ] box (works)
  - [ ] sub check [ ] box (does not work)
- [ ] check [ ] box (works)
```

Reply:
> - [ ] check [ ] box (works)
>  - [ ] sub check [ ] box (does not work)
> - [ ] check [ ] box (works)


I also cannot reproduce this. Is the sub-element supposed to be indented? Because on my build it behaves identical to the other two (including the checkbox behaviour)

Reply:
Yes, the second item should be indented. See this screenshot:

![Screenshot](https://user-images.githubusercontent.com/6277619/178160757-354f3edd-49e3-4ef8-8388-53d55f35d59e.png)